simplematrixbotlib/match.py

```diff
@@ -41,6 +43,34 @@ def is_from_userid(self, userid):
         """
         return self.event.sender == userid
 
+    def is_from_allowed_user(self):
+        """
+        Returns
+        -------
+        boolean
+            Returns True if the event was sent from an allowed userid
+        """
+        allowlist = self._bot.config.allowlist
+        blocklist = self._bot.config.blocklist
+        sender = self.event.sender
+        # if there is no explicit allowlist, default to allow
+        is_allowed = False if len(allowlist) > 0 else True
+
+        try:
+            for id in allowlist:
+                if re.fullmatch(id, sender):
+                    is_allowed = True
+                    break
+            for id in blocklist:
+                if re.fullmatch(id, sender):
+                    is_allowed = False
+                    break
+        except re.error as e:
```

Comment:
What about removing this error handling, and adding a check in the config loader?

Reply:
yes, that seems reasonable, I'll do that.